pkg/endpoint/public_ip.go

```diff
@@ -108,30 +111,44 @@ func publicIP(clientset kubernetes.Interface, namespace, value string) (string,
 }
 
 func publicLoadBalancerIP(clientset kubernetes.Interface, namespace, loadBalancerName string) (string, error) {
+	ip := ""
 	service, err := clientset.CoreV1().Services(namespace).Get(context.TODO(), loadBalancerName, metav1.GetOptions{})
 	if err != nil {
 		return "", errors.Wrapf(err, "error getting service %q for the public IP address", loadBalancerName)
 	}
 
-	if len(service.Status.LoadBalancer.Ingress) < 1 {
-		return "", errors.Errorf("service %q doesn't contain any LoadBalancer ingress", loadBalancerName)
+	retryConfig := wait.Backoff{
+		Cap:      2 * time.Minute,
+		Duration: 1 * time.Second,
+		Factor:   2,
+		Steps:    4,
 	}
 
-	ingress := service.Status.LoadBalancer.Ingress[0]
-	if ingress.IP != "" {
-		return ingress.IP, nil
-	}
-
-	if ingress.Hostname != "" {
-		ip, err := publicDNSIP(clientset, namespace, ingress.Hostname)
-		if err != nil {
-			return "", errors.Wrapf(err, "error resolving LoadBalancer ingress HostName %q", ingress.Hostname)
+	err = retry.OnError(retryConfig, func(err error) bool {
+		klog.Infof("Waiting for LoadBalancer to be ready: %s", err)
+		return true
+	}, func() error {
```

Comment:
Shouldn't you get the service here or am I missing something?

Reply:
I think so...

Reply:
Sorry for the update delay. @tpantelis thanks for catching this important "detail" ;) Without this it would have only worked waiting for the dns resolution, not the ingress.